queue/private.php

```diff
@@ -1,141 +1,164 @@
+			</div>
+			<div class="client-time"><b>Scheduled Appointment Time: </b>{{client.scheduledTime | date: "h:mm a"}}</div>
+			<div class="client-email" ng-if="client.emailAddress != null" ng-cloak>
+				<span><b>Email:</b> {{client.emailAddress}}</span>
 			</div>
+			<div class="client-phoneNumber" ng-if="client.phoneNumber != null" ng-cloak>
+				<span><b>Phone Number:</b> {{client.phoneNumber}}</span>
+			</div>
+		</div>
+
+		<div class="client-progress">
+			<h4>Update Progress:</h4>
+			<button type="button" 
+				class="wdn-button wdn-button-triad checkin" 
+				ng-show="!client.checkedIn"
+				ng-click="checkIn()">
+				Checked In
+			</button>
+			<button type="button"
+				class="wdn-button wdn-button-triad paperworkComplete" 
+				ng-show="client.checkedIn && !client.paperworkComplete"
+				ng-click="pwFilledOut()">
+				Completed Paperwork
+			</button>
+			<button type="button" 
+				class="wdn-button wdn-button-triad preparing" 
+				ng-show="client.paperworkComplete && !client.preparing"
+				ng-click="nowPreparing()">
+				Preparing
+			</button>
+			<select ng-show="client.preparing"
+				ng-model="client.selectedStationNumber" 
+				ng-options="stationNumber for stationNumber in stationNumbers">
+				<option value="" style="display:none;">-- Select Station --</option>
+			</select>
+			<button type="button" 
+				class="wdn-button wdn-button-triad ended" 
+				ng-show="client.preparing"
+				ng-disabled="client.selectedStationNumber == null"
+				ng-click="completeAppointment()">
+				Finished
+			</button>
 		</div>
+
+		<div class="greeter-directions">
+			<h4>Instructions:</h4>
+			<p>
+				Once a client has completed a step, click on the corresponding button. This will log the time at which each step is completed. PROGRESS CANNOT BE UNDONE, so
+				be sure to verify that the step is fully completed before clicking to progress the client. If, for any reason, a client does not complete their appointment,
+				fill out the form below and explain (in 255 characters or less) why the appointment was not completed. Appointments will disappear shortly after being marked
+				complete or incomplete.
+			</p>
+		</div>
+		
+		<div class="appointment-not-complete">
+			<form>
+				<h4>Appointment Not Completed:</h4>
+				<label>Explain why the appointment was not completed.</label>
+				<textarea ng-model="explanation" 
+					placeholder="-- Expanation --" 
+					class="form-control" 
+					cols="300" 
+					rows="3" 
+					maxlength="255" 
+					ng-maxlength="255">
+				</textarea>
+				<span class="wdn-pull-right">{{ explanation ? explanation.length : 0 }}/255</span>
+				<button class="wdn-button wdn-button-brand" 
+					ng-show="client.checkedIn" 
+					ng-click="incompleteAppointment(explanation)">
+					Submit Incomplete Appointment
+				</button>
+				<button class="wdn-button wdn-button-brand" 
+					ng-show="!client.checkedIn" 
+					ng-click="cancelledAppointment()">
+					Submit Cancelled Appointment
```

Comment:
I just realized that one can hit this button a bunch of times and it will create a new entry in ServicedAppointment each time someone hits this button... I suppose we could add a line into the PHP that ensures there isn't already an entry? But perhaps we should (perhaps in addition) add in something on the front-end that prevents someone from hitting this button if the appointment has ended? Same thing with the incompleteAppointment button?

Reply:
Definitely. I'll look into it.

Reply:
Done.